### PR TITLE
CF-300 Fixed matching src image and dst image

### PR DESCRIPTION
**Pull request by ntymtsiv**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:51: [W0212(protected-access), 
GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2776/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:51: [W0212(protected-access), 
GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def get_matching_image(self, uuid, name, size, checksum):
             image_matches = (
                 img.id == uuid or
                 img.name == name and
-                img.size == size and
+                str(img.size) == str(size) and
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Why? What are the cases when `img.size == size` breaks?

Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: I guess better would be to apply following patch:
```diff
diff --git a/cloudferrylib/os/storage/cinder_storage.py b/cloudferrylib/os/storage/cinder_storage.py
index 6c6afd8..1725163 100644
--- a/cloudferrylib/os/storage/cinder_storage.py
+++ b/cloudferrylib/os/storage/cinder_storage.py
@@ -353,7 +353,7 @@ class CinderStorage(storage.Storage):
                 'image_id': vol.volume_image_metadata['image_id'],
                 'checksum': vol.volume_image_metadata['checksum'],
                 'image_name': vol.volume_image_metadata.get('image_name'),
-                'size': vol.volume_image_metadata.get('size')
+                'size': int(vol.volume_image_metadata.get('size', 0))
             }
         if cfg.storage.backend == utils.CEPH:
             volume['path'] = "%s/%s%s" % (
```

File: tests/cloudferrylib/os/image/test_glance_image.py [hunk 2]
```diff
@@ -253,3 +253,30 @@ def test_read_info_default(self):
...
+        self.assertEqual(self.fake_image_1, img)
+
+    def test_get_matching_image_negative(self):
+        fake_images = [self.fake_image_1, self.fake_image_2]
+        self.glance_mock_client().images.list.return_value = fake_images
+
+        img = self.glance_image.get_matching_image(
+            uuid='mismatching', checksum='mismatch_checksum',
+            size=2048, name='mismatch_name')
+        self.assertEqual(None, img)
```
Comment (antonf) [inline tests/cloudferrylib/os/image/test_glance_image.py]: Please also add test like

         fake_images = [self.fake_image_1, self.fake_image_2]
         self.glance_mock_client().images.list.return_value = fake_images

         img = self.glance_image.get_matching_image(
             uuid='mismatching', checksum=self.fake_image_1.checksum,
             size='000' + str(self.fake_image_1.size), name=self.fake_image_1.name)
         self.assertEqual(self.fake_image_1, img)

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:140: [W0221(arguments-differ), 
CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:380: [W0613(unused-argument), 
CinderStorage.convert_snapshot] Unused argument 'cloud'
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:51: [W0212(protected-access), 
GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1, can be merged

**Pull request merged**